Homeworks/module_14/test_hw.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from Homeworks.module_14.pages import MainPage, LoggedMainPage
+
+
+def test_login(init_driver, user_test_data):
+    main_page = MainPage(init_driver)
+    login_page = main_page.open_login()
+
+    try:
+        login_page.login_field
+    except Exception as e:
+        pytest.fail(f"Username element wasn't found: {e}")
+
+    try:
+        login_page.password_field
+    except Exception as e:
+        pytest.fail(f"Password element wasn't found: {e}")
+
+    logged_main_page = login_page.login(*user_test_data)
+
+    try:
```

Comment:
it would be better use `assert {expr}, {error_message}` expression instead of try-except blocks, eg:

`assert logged_main_page.header.user_field.get_text() == f'Welcome {user_test_data[0]}', "Username element wasn't found"`

Reply:
Updated. But if the element doesn't exist then the error log just contains Selenium exception